pkg/controllers/disruption/pdblimits.go

```diff
@@ -56,10 +59,16 @@ func NewPDBLimits(ctx context.Context, kubeClient client.Client) (*PDBLimits, er
 
 // CanEvictPods returns true if every pod in the list is evictable. They may not all be evictable simultaneously, but
 // for every PDB that controls the pods at least one pod can be evicted.
+// nolint:gocyclo
 func (s *PDBLimits) CanEvictPods(pods []*v1.Pod) (client.ObjectKey, bool) {
 	for _, pod := range pods {
+		// If the pod isn't an "active" pod or isn't eligible for being evicted, then a fully blocking PDB doesn't matter
+		// This is due to the fact that we won't call the eviction API on these pods when we are disrupting the node
+		if !podutil.IsActive(pod) || !podutil.IsEvictable(pod, time.Now()) {
```

Comment:
commenting here again just to make sure it's caught, I wonder if there's a way we can just encapsulate the IsActive() call here within the IsEvictable call, so that we don't have two interpretations of the same helper.

Reply:
Good callout. Changed the `IsEvictable` function. You've convinced me that we should just have an `IsEvictable` and `IsWaitingEviction` function